src/geojson_types.jl

```diff
@@ -0,0 +1,184 @@
+abstract type GeoJSONT{D} end
+abstract type AbstractGeometry{D} <: GeoJSONT{D} end
+abstract type AbstractFeatureCollection{D} <: GeoJSONT{D} end
+
+struct Point{D} <: AbstractGeometry{D}
+    type::String
```

Comment:
This could be particularly costly in memory because we are allocating a string for every single point. Is there a way to throw that information away, and get it back from a function of the type?

We could also use https://github.com/JuliaString/InternedStrings.jl to avoid the allocations